src/EnergyPlus/HVACVariableRefrigerantFlow.cc

```diff
@@ -5246,41 +5102,548 @@ namespace HVACVariableRefrigerantFlow {
+                            ShowContinueError("  use a Setpoint Manager to establish a setpoint at the coil control node.");
+                            SetPointErrorFlag = true;
+                        } else {
+                            bool SPNotFound = false;
+                            EMSManager::CheckIfNodeSetPointManagedByEMS(
+                                VRFTU(TUIndex).VRFTUOutletNodeNum, EMSManager::iTemperatureSetPoint, SetPointErrorFlag);
+                            SPNotFound = SPNotFound || SetPointErrorFlag;
+                            EMSManager::CheckIfNodeSetPointManagedByEMS(
+                                VRFTU(TUIndex).coolCoilAirOutNode, EMSManager::iTemperatureSetPoint, SetPointErrorFlag);
+                            SPNotFound = SPNotFound || SetPointErrorFlag;
+                            EMSManager::CheckIfNodeSetPointManagedByEMS(
+                                VRFTU(TUIndex).heatCoilAirOutNode, EMSManager::iTemperatureSetPoint, SetPointErrorFlag);
+                            SPNotFound = SPNotFound || SetPointErrorFlag;
+                            if (SPNotFound) {
+                                ShowSevereError("ZoneHVAC:TerminalUnit:VariableRefrigerantFlow: Missing temperature setpoint for unitary system = " +
+                                                VRFTU(TUIndex).Name);
+                                ShowContinueError("  use a Setpoint Manager to establish a setpoint at the coil control node.");
+                                ShowContinueError("  or use an EMS actuator to establish a temperature setpoint at the coil control node.");
+                            }
+                        }
                     }
 
-                    if (CheckZoneEquipmentList(DataHVACGlobals::cVRFTUTypes(VRFTU(TUIndex).VRFTUType_Num), VRFTU(TUIndex).Name)) continue;
-                    ShowSevereError("InitVRF: VRF Terminal Unit = [" + DataHVACGlobals::cVRFTUTypes(VRFTU(TUIndex).VRFTUType_Num) + ',' + VRFTU(TUIndex).Name +
-                                    "] is not on any ZoneHVAC:EquipmentList.  It will not be simulated.");
+                    if (AirLoopFound || CheckZoneEquipmentList(DataHVACGlobals::cVRFTUTypes(VRFTU(TUIndex).VRFTUType_Num), VRFTU(TUIndex).Name)) continue;
+                    ShowSevereError("InitVRF: VRF Terminal Unit = [" +DataHVACGlobals::cVRFTUTypes(VRFTU(TUIndex).VRFTUType_Num) + ',' + VRFTU(TUIndex).Name +
+                                    "] is not on any ZoneHVAC:EquipmentList, AirloopHVAC or AirLoopHVAC:OutdoorAirSystem:EquipmentList.  It will not "
+                                    "be simulated.");
                     ShowContinueError("...The VRF AC System associated with this terminal unit may also not be simulated.");
                 }
-            } // IF(ZoneEquipInputsFilled) THEN
+            }
+
+            // TU inlet node must be the same as a zone exhaust node and the OA Mixer return node
```

Comment:
Some checks from GetInput were moved to Init, and inside a one-time IF block so as not to slow down simulation.